transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/WsnConnectorRule.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright (c) 2007-2014, Kaazing Corporation. All rights reserved.
+ */
+
+package org.kaazing.gateway.transport.wsn;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+import org.kaazing.gateway.resource.address.ResourceAddress;
+import org.kaazing.gateway.resource.address.ResourceAddressFactory;
+import org.kaazing.gateway.transport.BridgeServiceFactory;
+import org.kaazing.gateway.transport.TransportFactory;
+import org.kaazing.gateway.transport.http.HttpConnector;
+import org.kaazing.gateway.transport.nio.NioSocketAcceptor;
+import org.kaazing.gateway.transport.nio.NioSocketConnector;
+import org.apache.log4j.PropertyConfigurator;
+import org.apache.mina.core.future.ConnectFuture;
+import org.apache.mina.core.service.IoHandler;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+import org.kaazing.gateway.util.scheduler.SchedulerProvider;
+
+
+/**
+ * Declaring an instance of this class as a @Rule causes the gateway to be
+ * started in process before each test method and stopped after it. The rule
+ * can be chained with a K3poRule for use with robot (this causes Robot to be
+ * started before the gateway and stopped after it).
+ */
+public class WsnConnectorRule implements TestRule {
+
+    private final String log4jPropertiesResourceName;
+    private ResourceAddressFactory addressFactory;
+    private WsnConnector wsnConnector;
+
+
+    @Override
+    public Statement apply(Statement base, Description description) {
+        return new ConnectorStatement(base);
+    }
+
+    public WsnConnectorRule() {
+        this(null);
+    }
+
+    public WsnConnectorRule(String log4jPropertiesResourceName) {
+        this.log4jPropertiesResourceName = log4jPropertiesResourceName;
+    }
+
+    public ConnectFuture connect(String connect, Long wsInactivityTimeout, IoHandler connectHandler)
+            throws InterruptedException {
+        Map<String, Object> connectOptions = new HashMap<>();
+        if (wsInactivityTimeout != null) {
+            connectOptions.put("inactivityTimeout", wsInactivityTimeout);
+        }
+        final ResourceAddress connectAddress =
+                addressFactory.newResourceAddress(URI.create(connect), connectOptions);
+
+        ConnectFuture future = wsnConnector.connect(connectAddress, connectHandler, null);
+
+        future.await(TimeUnit.MILLISECONDS.toMillis(3000));
```

Comment:
I guess I just followed WsebConnectorRule ! But this makes sense. I modified WsnConnectorRule to return future and the test can decide how to wait and get the values. We evolve further in future as the connect method may need some more arguments